src/components/Admin/volunteerstatstable.tsx

```diff
@@ -1,107 +1,217 @@
+import { useMemo, useState } from 'react'
+import { useQueryClient } from '@tanstack/react-query'
+import { Timestamp } from 'firebase/firestore'
+import { SubmitHandler } from 'react-hook-form'
+
+import Button from '@/components/UI/button'
+import Card from '@/components/UI/card'
+import Form from '@/components/UI/FormComponents/Form'
+import TextField from '@/components/UI/FormComponents/TextField'
+import Spinner from '@/components/UI/loadingSpinner'
+import { useVolunteerStatsByDateRange } from '@/hooks/admin'
+import { formatTimestamp, requiredMessage } from '@/utils'
+
+interface FormValues {
+  fromTime: string
+  toTime: string
+}
+
+const YEAR_IN_MS = 31556952000
+const queryKey = 'mainVolunteerStatsTable'
+
 const headers = [
-  'Number of Clients',
   'Number of Visits',
   'Walk Duration (mins)',
   'Walk Distance (kms)',
   'Commute Distance (kms)'
 ]
 
-const data = {
-  voluteerCount: 10,
-  totalClientCount: 50,
-  totalCommuteDistance: 50,
-  totalVisitCount: 500,
-  totalWalkDistance: 500,
-  totalWalkTime: 5000,
-  avgClientCount: 50,
-  avgCommuteDistance: 50,
-  avgVisitCount: 500,
-  avgWalkDistance: 500,
-  avgWalkTime: 5000
-}
+const VolunteerStatsTable = () => {
+  const queryClient = useQueryClient()
+  const now = Date.now()
+
+  const start = Timestamp.fromMillis(now - YEAR_IN_MS)
+  const end = Timestamp.fromMillis(now)
 
-function VoluteerStatsTable() {
-  const {
-    voluteerCount,
-    totalClientCount,
-    totalCommuteDistance,
-    totalVisitCount,
-    totalWalkDistance,
-    totalWalkTime,
-    avgClientCount,
-    avgCommuteDistance,
-    avgVisitCount,
-    avgWalkDistance,
-    avgWalkTime
-  } = data
+  const [from, setFrom] = useState<Timestamp>(start)
+  const [to, setTo] = useState<Timestamp>(end)
+
+  const { isLoading, data: volunteerStats } = useVolunteerStatsByDateRange(
+    queryKey,
+    from,
+    to
+  )
 
   const totalStats = [
-    totalClientCount,
-    totalVisitCount,
-    totalWalkTime,
-    totalWalkDistance,
-    totalCommuteDistance
+    volunteerStats?.totalVisitCount,
+    volunteerStats?.totalWalkTime,
+    volunteerStats?.totalWalkDistance,
+    volunteerStats?.totalCommuteDistance
   ]
 
   const avgStats = [
-    avgClientCount,
-    avgVisitCount,
-    avgWalkTime,
-    avgWalkDistance,
-    avgCommuteDistance
+    volunteerStats?.avgVisitCount,
+    volunteerStats?.avgWalkTime,
+    volunteerStats?.avgWalkDistance,
+    volunteerStats?.avgCommuteDistance
   ]
 
+  const refetchStats = () => {
+    // ! have to remove queries as staleTime and cacheTimes are infinite.
```

Comment:
Do queries persist in local storage? Maybe it's fine to cache them for the entire session.

Reply:
Yes, they do, I just thought that it might be potentially bad when the queries do get big. I will change them for now and see how it goes.